core/src/main/java/io/fabric8/maven/core/util/ResourceVersioning.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2016 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+package io.fabric8.maven.core.util;
+
+/**
+ * @author nicola
+ * @since 24/05/2017
+ */
+public class ResourceVersioning {
+
+    private String coreVersion;
+
+    private String extensionsVersion;
+
+    private String appsVersion;
+
+    public ResourceVersioning() {
```

Comment:
Good idea to encapsulate the versioning in an own class. However these are only a subset of APIs available. We should provide the most important as typed versions, but also allow for extension for other APIs like `withApiVersion(String apiName, String version)` so that it could be easily extended further.